consensus/ibft/pos.go

```diff
@@ -0,0 +1,196 @@
+package ibft
+
+import (
+	"errors"
+	"fmt"
+	"github.com/0xPolygon/polygon-sdk/contracts/staking"
+	"github.com/0xPolygon/polygon-sdk/types"
+)
+
+// PoSMechanism defines specific hooks for the Proof of Stake IBFT mechanism
+type PoSMechanism struct {
+	// Reference to the main IBFT implementation
```

Comment:
I like to enforce the practice of documenting every section of a structure we're using. It would seem messy to just document only certain parts of it, and leave others undocumented, even if they're clear in their use

Reply:
again, not sure how much value these comments bring, but in my experience, people just ignore these comments. Furthermore, if any time in the future the type of the field struct changes, people tend to forget to change the comments... Anyway, this Is ok for now.